web/src/main/java/se/inera/intyg/webcert/web/service/facade/patient/GetPatientFacadeServiceImpl.java

```diff
@@ -0,0 +1,82 @@
+import se.inera.intyg.schemas.contract.InvalidPersonNummerException;
+import se.inera.intyg.schemas.contract.Personnummer;
+import se.inera.intyg.webcert.web.service.monitoring.MonitoringLogService;
+import se.inera.intyg.webcert.web.web.controller.facade.PatientController;
+import se.inera.intyg.webcert.web.web.controller.facade.dto.PatientResponseDTO;
+
+@Service
+public class GetPatientFacadeServiceImpl implements GetPatientFacadeService {
+    private static final Logger LOG = LoggerFactory.getLogger(PatientController.class);
+
+    final private PUService puService;
+
+    final private MonitoringLogService monitoringService;
+
+    @Autowired
+    public GetPatientFacadeServiceImpl(PUService puService, MonitoringLogService monitoringService) {
+        this.puService = puService;
+        this.monitoringService = monitoringService;
+    }
+
+    @Override
+    public PatientResponseDTO getPatient(String patientId) {
+        try {
+            Personnummer formattedPatientId = formatPatientId(patientId);
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Getting patient info for: {}", formattedPatientId.getPersonnummerHash());
+            }
+
+            PersonSvar personSvar = puService.getPerson(formattedPatientId);
+
+            monitoringService.logPULookup(formattedPatientId, personSvar.getStatus().name());
+
+            return convertPatientResponse(personSvar, patientId);
+
+        } catch (InvalidPersonNummerException e) {
+            LOG.error(e.getMessage());
+            return null;
+        }
+    }
+
+    private PatientResponseDTO convertPatientResponse(PersonSvar personSvar, String patientId) {
+        final var patient = Patient.builder()
+                .personId(
+                        PersonId.builder()
+                                .id(patientId)
+                                .type("")
+                                .build()
+                )
+                .firstName(personSvar.getPerson().getFornamn())
+                .lastName(personSvar.getPerson().getEfternamn())
+                .middleName(personSvar.getPerson().getMellannamn())
+                .fullName(getFullName(personSvar))
+                .deceased(personSvar.getPerson().isAvliden())
+                .protectedPerson(personSvar.getPerson().isSekretessmarkering())
+                .testIndicated(personSvar.getPerson().isTestIndicator())
+                .build();
+
+        return PatientResponseDTO.create(patient, personSvar.getStatus());
+    }
+
+    private String getFullName(PersonSvar personSvar) {
+        if(personSvar.getPerson().getMellannamn() == null || personSvar.getPerson().getMellannamn().length() == 0) {
+            return personSvar.getPerson().getFornamn() + " " + personSvar.getPerson().getEfternamn();
+        }
+        return personSvar.getPerson().getFornamn() + " " + personSvar.getPerson().getMellannamn() + " " + personSvar.getPerson().getEfternamn();
```

Comment:
The code would be easier to read/process if we introduce a variable for `personSvar.getPerson()`
```suggestion
        final var person = personSvar.getPerson();
        if (person.getMellannamn() == null || person.getMellannamn().length() == 0) {
            return person.getFornamn() + " " + person.getEfternamn();
        }
        return person.getFornamn() + " " + person.getMellannamn() + " " + person.getEfternamn();
```